### PR TITLE
Fix PDAE system example and derivative neural network approximation

### PR DESCRIPTION
**Pull request by KirillZubov**
https://github.com/SciML/NeuralPDE.jl/issues/302

### CODE REVIEW

File: docs/src/pinn/system.md [hunk 1]
```diff
@@ -80,26 +82,156 @@ ts,xs = [infimum(d.domain):0.01:supremum(d.domain) for d in domains]
...
+```
+
+![sol_uq1](https://user-images.githubusercontent.com/12683885/122979254-03634e80-d3a0-11eb-985b-d3bae2dddfde.png)
+
+![sol_uq2](https://user-images.githubusercontent.com/12683885/122979278-09592f80-d3a0-11eb-8fee-de3652f138d8.png)
+
+![sol_uq3](https://user-images.githubusercontent.com/12683885/122979288-0e1de380-d3a0-11eb-9005-bfb501959b83.png)
+
+
+## Derivative neural network approximation
```
Comment (ChrisRackauckas) [inline docs/src/pinn/system.md:107]: what does that mean?

Comment (KirillZubov) [reply docs/src/pinn/system.md:107]: we approximate the derivative of the neural network with another neural network `Dt(u1(t,x)) ~ Dtu1(t,x)` thus we avoid taking the second derivative `Dt(Dtu1(t,x)) `.
Since `u3` is only in the first and second equations, that its accuracy during training is determined by the accuracy of the second numerical derivative `Dtt(u1(t,x)) ~ Dxx(u1(t,x)) + u3(t,x)*sin(pi*x)`.
So this is such the trick to not take the second numeric derivative that allows an increase in the marginal precision for optimization.

Comment (ChrisRackauckas) [reply docs/src/pinn/system.md:107]: ahh this needs more of an explanation.

Comment (KirillZubov) [reply docs/src/pinn/system.md:107]: https://github.com/SciML/NeuralPDE.jl/blob/Systems_of_PDE_fix/docs/src/pinn/system.md#comparison-of-the-second-numerical-derivative-and-numerical--neural-network-derivative

Comment (ChrisRackauckas) [reply docs/src/pinn/system.md:107]: But there's no explanation. There's a title, and then there's no text explaining what the section is actually about. There needs to be some introductory text as to what you're doing there and why.

Comment (KirillZubov) [reply docs/src/pinn/system.md:107]: oh, I have thought you need more explanation :) 
ok, I will add a text explanation to the doc.

**Comment by KirillZubov** [comment]
I used to think that the problem with the systems is that the gradient is not updating correctly and therefore the third equation had a much higher error. It turned out that this was due to a derivative error. The error grows very quickly with the degree of the numerical derivative.
Therefore, we definitely need to update the calc of the numerical derivative and achieve greater accuracy and stability for order > 1 .

**Comment by ChrisRackauckas** [comment]
Yeah, we just need Diffractor.jl done for that. Keno has examples.

**Comment by KirillZubov** [comment]
And the second, regarding the adaptive reweighting,  the problem is not only that the derivatives have frequency scale amplification but also that each degree of the derivative has a large error, which clearly interferes with the gradient descent.

I guess instability gives a splash of the gradient.
max(grad(loss(higher_order_derivatives(u)+big error ))) >>  max(grad(loss(lower-order_derivatives(u))))

but this is only for numeric derivatives, automatic differentiation does not have such problems, as I understand it, although.

**Comment by ChrisRackauckas** [comment]
Also the way the derivative is calculated here isn't great, it should probably use FiniteDifferences.jl if anything because you don't want to nest 1st derivatives and instead adapt the epsilon and all of that.
